protocols/virtualfund/virtualfund.go

```diff
@@ -297,19 +297,15 @@ func (connection *Connection) insertExpectedGuarantees(a0 types.Funds, b0 types.
 func (s VirtualFundObjective) fundingComplete() bool {
 
 	// Each peer commits to an update in L_{i-1} and L_i including the guarantees G_{i-1} and {G_i} respectively, and deducting b_0 from L_{I-1} and a_0 from L_i.
-	// A = P_0 and B=P_n+1 are special cases. A only does the guarantee for L_0 (deducting a0), and B only foes the guarantee for L_n (deducting b0).
-
-	n := s.n
+	// A = P_0 and B=P_n are special cases. A only does the guarantee for L_0 (deducting a0), and B only foes the guarantee for L_n (deducting b0).
 
 	switch {
-	case s.MyRole == 0: // Alice
+	case s.isAlice(): // Alice
 		return s.ToMyRight.ledgerChannelAffordsExpectedGuarantees()
-	case s.MyRole < n+1: // Intermediary
+	default: // Intermediary
 		return s.ToMyRight.ledgerChannelAffordsExpectedGuarantees() && s.ToMyLeft.ledgerChannelAffordsExpectedGuarantees()
-	case s.MyRole == n+1: // Bob
+	case s.isBob(): // Bob
 		return s.ToMyLeft.ledgerChannelAffordsExpectedGuarantees()
-	default: // Invalid
-		return false
```

Comment:
We don't need to have this default case here because we prevent invalid roles in the constructor.